pkg/api/secretbootstrap/secretboostrap_test.go

```diff
@@ -209,6 +218,167 @@ func TestLoadConfigFromFile(t *testing.T) {
+			}
+
+			outFile = filepath.Join("testdata", "roundtrip_out.yaml")
+			err = SaveConfigToFile(outFile, c)
+			if err != nil {
+				t.Fatalf("error saving config file: %v", err)
+			}
+
+			inFile := filepath.Join("testdata", "zz_fixture_TestRoundtripConfig_basic_base.yaml")
+			in, _ := ioutil.ReadFile(inFile)
+			out, _ := ioutil.ReadFile(outFile)
+			if diff := cmp.Diff(in, out); diff != "" {
+				t.Fatalf("input and output configs are not equal. %s", diff)
+			}
+		})
+
+		t.Cleanup(func() {
+			if err := os.Remove(outFile); err != nil {
+				t.Fatalf("error removing output config file: %v", err)
+			}
+		})
+	}
+}
+
+func TestGroupClusters(t *testing.T) {
+	testCases := []struct {
+		name     string
+		input    SecretConfig
+		expected []SecretContext
+	}{
+		{
+			name: "no group",
+			input: SecretConfig{
+				From: map[string]ItemContext{
+					"item-a": {
+						Item:  "a",
+						Field: "field",
+					},
+				},
+				To: []SecretContext{{
+					Cluster:   "cluster1",
+					Namespace: "ns",
+					Name:      "a",
+				}},
+			},
+			expected: []SecretContext{{
+				Cluster:   "cluster1",
+				Namespace: "ns",
+				Name:      "a",
+			}},
+		},
+		{
+			name: "group",
+			input: SecretConfig{
+				From: map[string]ItemContext{
+					"item-a": {
+						Item:  "a",
+						Field: "field",
+					},
+				},
+				To: []SecretContext{{
+					ClusterGroups: []string{"group-a"},
+					Cluster:       "cluster1",
+					Namespace:     "ns",
+					Name:          "a",
+				}},
+			},
+			expected: []SecretContext{{
+				ClusterGroups: []string{"group-a"},
+				Namespace:     "ns",
+				Name:          "a",
+			}},
+		},
+		{
+			name: "mix",
+			input: SecretConfig{
+				From: map[string]ItemContext{
+					"item-a": {
+						Item:  "a",
+						Field: "field",
+					},
+				},
+				To: []SecretContext{
+					{
+						ClusterGroups: []string{"group-a"},
+						Cluster:       "cluster1",
+						Namespace:     "ns",
+						Name:          "a",
+					},
+					{
+						Cluster:   "cluster2",
+						Namespace: "ns",
+						Name:      "b",
+					},
+				},
+			},
+			expected: []SecretContext{
+				{
+					ClusterGroups: []string{"group-a"},
+					Namespace:     "ns",
+					Name:          "a",
+				},
+				{
+					Cluster:   "cluster2",
+					Namespace: "ns",
+					Name:      "b",
+				},
+			},
+		},
+		{
+			name: "multiple groups",
+			input: SecretConfig{
+				From: map[string]ItemContext{
+					"item-a": {
+						Item:  "a",
+						Field: "field",
+					},
+				},
+				To: []SecretContext{{
+					ClusterGroups: []string{"group-a", "group-b"},
+					Cluster:       "cluster1",
+					Namespace:     "ns",
+					Name:          "a",
+				}},
+			},
+			expected: []SecretContext{{
+				ClusterGroups: []string{"group-a", "group-b"},
+				Namespace:     "ns",
+				Name:          "a",
+			}},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			sc := tc.input
+			sc.groupClusters()
+			if !reflect.DeepEqual(sc.To, tc.expected) {
```

Comment:
can we use the `diff := cmp.Diff(in, out); diff != "` pattern here too? I think all our new tests use `cmp` and not `reflect`.

Reply:
ahhh didn't think about using that here as well, thanks for the info!